src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -897,6 +898,10 @@ private JsValue ArchiveAt(JsValue self, JsValue[] args)
                         throw new InvalidOperationException($"Failed to fetch the metadata of document '{archivedDocId}'");
                     }
                     
+                    if(metadata.TryGetMember(Constants.Documents.Metadata.Archived, out _))
```

Comment:
Shouldn't we check document flags instead of metadata entry?

Reply:
done